### DISCUSSION TITLE
Linux: Using SAPI5 voices installed in WINE?

### DISCUSSION THREAD

**Discussion by hajonnes** [edrlab/thorium-reader#1813] [Ideas]
Hi,
I have very good SAPI5 voices installed in wine.
You have a window version hence an implemented way of communicating with SAPI5 voices.
Is it possible to use the SAPI5 voices installed in wine together with Thorium Reader for linux?

If not I would very much like to push for this. 
There are a lot of good voices for windows. To my knowledge (been using TTS since 2005) there are very limited options for linux.

All the best, and have a nice day!

/hajonnes

**Comment by danielweck** [comment]
Hello, Thorium discovers the available TTS voices via a standard web API in the Chromium open source web browser engine, which itself picks up the list of voices installed on the operating system. In other words, in the current implementation Thorium's own code has very limited control, we rely entirely on the standard web API.

**Reply by hajonnes** [reply]
Perhaps this is not your biggest priority,
but it would be really cool if you could support how to run Thorium in [wine](https://www.winehq.org/). I think we are many linux users that rely on TTS run in wine since the quality of voices are so much better for windows.

I got thorium partly working and I could select my voices but when I pressed "play" nothing happened. I got error about  dxgi and d3d11 but I do not know if they are the culpit or how to fix it. I do not have the skill to check out what is wrong.
I used wine as a 64bit prefix (windows envrionment) with SAPI 5 voices and native windowsfiles supplied by [winetricks](https://wiki.winehq.org/Winetricks) for:
speechsdk
vcrun2008
vcrun2010
msvcirt
d3dx9
mscorefonts

That comes from the setup I use to run balabolka tts in wine.

**Reply by danielweck** [reply]
This sounds like an interesting / useful problem to solve, but this appears to be something that needs to be fixed outside of Thorium, at the installation / integration level. Hopefully the user community can figure something out. I don't think that the core developer "team" (i.e. 1.0-1.5 full time equivalent, on average) will have the resources to tackle this.
